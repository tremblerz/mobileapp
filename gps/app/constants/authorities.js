export const PUBLIC_DATA_URL =
  'https://raw.githubusercontent.com/beoutbreakprepared/nCoV2019/master/latest_data/latestdata.csv';

export const AUTHORITIES_LIST_URL =
  'https://raw.githubusercontent.com/Path-Check/safeplaces-frontend/develop/healthcare-authorities.yaml';
