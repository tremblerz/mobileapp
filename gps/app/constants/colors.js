const colors = {
  BLACK: '#000',
  WHITE: '#FFF',
  DODGER_BLUE: '#428AF8',
  SILVER: '#BEBEBE',
  TORCH_RED: '#F8262F',
  MISCHKA: '#E5E4E6',
  PRIMARY_TEXT: '#000',
  SUCCESS: '#41dca4', // Green
  WARNING: '#ffc000', // Orange
  VIOLET_ALPHA_06: '#f7f8ff', // cannot use transparency
  VIOLET: '#4051DB',
  TRANSPARENT: 'transparent',

  // For notifications
  HIGHEST_RISK: '#e74c3c',
  MIDDLE_RISK: '#e74c3c',
  LOWER_RISK: '#e74c3c',
  LOWEST_RISK: '#e74c3c',

  REG_BUTTON: '#428AF8',
  POS_BUTTON: '#32A852',
  NEG_BUTTON: '#F8262F',
  SENS_BUTTON: '#6C3794',
  VIOLET_BUTTON: '#6979F8',
  VIOLET_BUTTON_DARK: '#3A4CD7',

  VIOLET_TEXT: '#3C4ED8',

  GRAY_BACKGROUND: '#DADADA',

  INTRO_WHITE_BG: '#F7F8FF',
  PULSE_WHITE: '#FFFFFF40',

  DIVIDER: '#A5AFFB',

  BLUE_BUTTON: '#05369B',
  BLUE_TO_BUTTON: '#185BD3',
  RED_BUTTON: '#E3554E',
  RED_TO_BUTTON: '#E97C79',
  GRAY_BUTTON: '#868686',
  GRAY_TO_BUTTON: '#868686',

  MAP_LINE_STROKE: '#4052DB',

  BOTTOM_SHEET_HEADER_INDICATOR: '#E1E4FF',

  BLUE_LINK: '#007AFF',

  RED_TEXT: '#FF5656',

  NAV_BAR_VIOLET: '#5061E6',
  NAV_BAR_BORDER: 'rgba(189, 195, 199,0.6)',

  DARK_GRAY: '#3C475B',
  LIGHT_GRAY: '#EEEEEE',

  MONO_DARK: '#202020',
  MONO_SECONDARY: '#757677',
};

export default colors;
