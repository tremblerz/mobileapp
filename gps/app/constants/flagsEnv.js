import env from 'react-native-config';

/**
 * Normalizes flags:
 *
 * Example:
 *
 * `{ flag_a: 'true' }` becomes `{ a: true }`
 *
 * @param {{[key: string]: string}} envConfig
 */
export function parseFlags(envConfig) {
  return Object.entries(envConfig)
    .filter(([key]) => key.toLowerCase().startsWith('flag'))
    .reduce((flags, [key, value]) => {
      const flag = key.replace(/^flag_/i, '');

      flags[flag] = value === 'true' || value === '1';
      return flags;
    }, {});
}

export const getBuildtimeFlags = () => parseFlags(env);
