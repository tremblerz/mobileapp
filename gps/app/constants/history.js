/**
 * Max esposure reporting window in days
 */
export const MAX_EXPOSURE_WINDOW_DAYS = 14;

/**
 * The value in minutes of each "bin" in the crossed path data.
 */
export const DEFAULT_EXPOSURE_PERIOD_MINUTES = 5;

/**
 * The value in minutes of how long an exposure at a location is
 *    considered concerning.
 */
export const CONCERN_TIME_WINDOW_MINUTES = 4 * 60; // 4 hours, in minutes

/**
 * The value in minutes of how frequently we should check intersection data if
 *    there has been no change to the authorities
 */
export const MIN_CHECK_INTERSECT_INTERVAL = 6 * 60;

/**
 * The value in minutes for the background task service to register for firing
 */
export const INTERSECT_INTERVAL = 60 * 12; // 12 Hours, the value is received in minutes

/**
 * Format of a single history item
 *
 * @typedef {{
 *   date: import("dayjs").Dayjs,
 *   exposureMinutes: number,
 * }} HistoryDay
 */

/**
 * Exposure history
 *
 * @typedef {!HistoryDay[]} History
 */
