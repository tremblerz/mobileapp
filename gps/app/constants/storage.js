export const LOCATION_DATA = 'LOCATION_DATA';
export const CONTACT_DATA = 'CONTACT_DATA';
export const PARTICIPATE = 'PARTICIPATE';
export const MY_UUIDs = 'MY_UUIDs';
export const CROSSED_PATHS = 'CROSSED_PATHS';
export const LANG_OVERRIDE = 'LANG_OVERRIDE';
export const DEBUG_MODE = 'DEBUG_MODE';
export const AUTHORITY_NEWS = 'AUTHORITY_NEWS';
export const LAST_CHECKED = 'LAST_CHECKED';
export const AUTHORITY_SOURCE_SETTINGS = 'AUTHORITY_SOURCE_SETTINGS';
export const ENABLE_HCA_AUTO_SUBSCRIPTION = 'ENABLE_HCA_AUTO_SUBSCRIPTION';
export const FEATURE_FLAG_VALS = 'FEATURE_FLAG_VALS';
