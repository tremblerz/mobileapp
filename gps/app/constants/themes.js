import styled from '@emotion/native';
import { ThemeProvider } from 'emotion-theming';
import React from 'react';

import Color from './colors';

/** Violet on pale violet bg. e.g. Settings */
export const defaultTheme = {
  background: Color.VIOLET_ALPHA_06,
  textPrimaryOnBackground: Color.VIOLET,
  textSecondaryOnBackground: 'rgba(64, 81, 219, 0.6)',

  navBar: Color.VIOLET,
  onNavBar: Color.WHITE,

  /** E.g. button bg */
  primary: Color.VIOLET_BUTTON,
  /** E.g. button text color */
  onPrimary: Color.WHITE,

  success: Color.SUCCESS,
  warning: Color.WARNING,
  disabled: 'rgba(64, 81, 219, 0.6)',
  border: Color.DIVIDER,
};

/** White on violet bg. E.g. Main screen */
export const violet = {
  ...defaultTheme,
  background: `linear-gradient(245.21deg, ${Color.VIOLET_BUTTON} 0%, ${Color.VIOLET_BUTTON_DARK} 100%);`,
  textPrimaryOnBackground: Color.WHITE,
  textSecondaryOnBackground: 'rgba(255, 255, 255, 0.6)',

  navBarBorder: Color.VIOLET, // no visible border

  primary: Color.WHITE,
  onPrimary: Color.VIOLET,
  disabled: '#ececec',
};

/** White on gray bg. E.g. Possible exposure mode on default screen */
export const charcoal = {
  ...defaultTheme,
  background: Color.DARK_GRAY,
  textPrimaryOnBackground: Color.WHITE,
  textSecondaryOnBackground: Color.LIGHT_GRAY,

  navBar: Color.DARK_GRAY,
  navBarBorder: Color.DARK_GRAY, // no visible border

  primary: Color.WHITE,
  onPrimary: Color.DARK_GRAY,

  disabled: '#A9AFBA',
};

const THEME_MAP = {
  default: defaultTheme,
  violet,
  charcoal,
};

/** A view which uses the theme's background color */
export const ThemedBackground = styled.View`
  background-color: ${({ theme }) => theme.background};
`;

/**
 * Return theme config for a known theme name
 *
 * @param {'default' | 'violet' | 'charcoal'} themeName
 */
export const getTheme = themeName => {
  const themeColors = THEME_MAP[themeName];
  if (!themeColors) {
    throw new Error(`Unknown theme ${themeName}`);
  }
  return themeColors;
};

/**
 * Configures themeable sub-components with background, text (primary,
 * secondary), button and other various theme colors.
 *
 * Usage:
 *
 * This will render a charcoal view, with a white heading, and a grey paragraph.
 *
 * ```jsx
 * <Theme use="charcoal" setBackground>
 *   <Typography use="headline2">Test</Typogrphy>
 *   <Typography secondary>Paragraph here</Typogrphy>
 * </Theme>
 * ```
 *
 * It makes available the following `props.theme.x` properties in styled components:
 *
 * - background - the standard bg color
 * - primary - the primary theme/button color on top of bakground (e.g. violet)
 * - onPrimary - color to show on top of primary (usually white)
 * - onPrimary - color to show on top of primary (usually white)
 * - textPrimaryOnBackground - text color to show on background
 * - textSecondaryOnBackground - secondary text color to show on background
 * - warning - warning color (always orange)
 * - success - success color (always green)
 * - border - border/divider color (usually gray)
 *
 * @param {{
 *   use?: string,
 *   setBackground?: boolean,
 * }} param0
 */
export const Theme = ({ use = 'default', setBackground = false, children }) => {
  return (
    <ThemeProvider theme={getTheme(use)}>
      {setBackground ? (
        <ThemedBackground>{children}</ThemedBackground>
      ) : (
        children
      )}
    </ThemeProvider>
  );
};

/**
 * @param {{theme: defaultTheme, secondary?: boolean}} param0
 */
export const themeTextColor = ({ theme, secondary }) =>
  secondary ? theme.textSecondaryOnBackground : theme.textPrimaryOnBackground;
