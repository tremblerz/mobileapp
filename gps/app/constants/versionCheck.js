export const APPSTORE_URL = 'itms-apps://itunes.apple.com/us/app/id1508266966';
export const PLAYSTORE_URL = 'market://details?id=org.pathcheck.covidsafepaths';
export const VERSION_URL =
  'https://rawcdn.githack.com/Path-Check/covid-safe-paths/master/versions.yaml';
export const YAML_MANDATORY_VERSION_KEY = 'mandatory upgrade version';
export const YAML_LATEST_VERSION_KEY = 'upgrade version';
