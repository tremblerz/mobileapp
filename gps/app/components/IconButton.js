import styled from '@emotion/native';
import React from 'react';
import { SvgXml } from 'react-native-svg';

import { themeTextColor } from '../constants/themes';

/**
 * A theme aware icon button
 *
 * @param {{
 *   icon: SvgXml;
 *   accessibilityLabel: string;
 *   secondary?: boolean;
 *   size?: number;
 * }} param0
 */
export const IconButton = ({
  icon,
  accessibilityLabel,
  secondary,
  size,
  ...otherProps
}) => {
  return (
    <Container accessibilityLabel={accessibilityLabel} {...otherProps}>
      <Icon
        xml={icon}
        secondary={secondary}
        width={size || 24}
        height={size || 24}
      />
    </Container>
  );
};

const Container = styled.TouchableOpacity`
  align-items: center;
  align-content: center;
`;

const Icon = styled(SvgXml)`
  color: ${themeTextColor};
`;
