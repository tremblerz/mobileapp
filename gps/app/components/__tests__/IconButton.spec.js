import { render } from '@testing-library/react-native';
import React from 'react';

import { Theme } from '../../constants/themes';
import { IconButton } from '../IconButton';

const svg = `<g />`;

it('renders the icon in a touchable opacity', () => {
  const { asJSON } = render(
    <Theme>
      <IconButton icon={svg} accessibilityLabel='Label' />
    </Theme>,
  );

  expect(asJSON()).toMatchSnapshot();
});

it('allows size override', () => {
  const { asJSON } = render(
    <Theme>
      <IconButton icon={svg} size={48} />
    </Theme>,
  );

  expect(asJSON()).toMatchSnapshot();
});

it('renders different color for secondary', () => {
  const { asJSON } = render(
    <Theme>
      <IconButton icon={svg} secondary />
    </Theme>,
  );

  expect(asJSON()).toMatchSnapshot();
});

it('changes color based on theme', () => {
  const { asJSON } = render(
    <>
      <Theme use='charcoal'>
        <IconButton icon={svg} label='charcoal' />
      </Theme>
      <Theme use='default'>
        <IconButton icon={svg} label='default' />
      </Theme>
    </>,
  );

  expect(asJSON()).toMatchSnapshot();
});
