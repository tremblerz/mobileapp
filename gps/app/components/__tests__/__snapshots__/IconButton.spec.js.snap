// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`allows size override 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    accessible={true}
    focusable={false}
    isTVSelectable={true}
    style={
      Object {
        "alignContent": "center",
        "alignItems": "center",
        "opacity": 1,
      }
    }
  >
    <SvgXml
      height={48}
      style={
        Object {
          "color": "#4051DB",
        }
      }
      width={48}
      xml="<g />"
    />
  </View>
</View>
`;

exports[`changes color based on theme 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    accessible={true}
    focusable={false}
    isTVSelectable={true}
    style={
      Object {
        "alignContent": "center",
        "alignItems": "center",
        "opacity": 1,
      }
    }
  >
    <SvgXml
      height={24}
      style={
        Object {
          "color": "#FFF",
        }
      }
      width={24}
      xml="<g />"
    />
  </View>
  <View
    accessible={true}
    focusable={false}
    isTVSelectable={true}
    style={
      Object {
        "alignContent": "center",
        "alignItems": "center",
        "opacity": 1,
      }
    }
  >
    <SvgXml
      height={24}
      style={
        Object {
          "color": "#4051DB",
        }
      }
      width={24}
      xml="<g />"
    />
  </View>
</View>
`;

exports[`renders different color for secondary 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    accessible={true}
    focusable={false}
    isTVSelectable={true}
    style={
      Object {
        "alignContent": "center",
        "alignItems": "center",
        "opacity": 1,
      }
    }
  >
    <SvgXml
      height={24}
      secondary={true}
      style={
        Object {
          "color": "rgba(64, 81, 219, 0.6)",
        }
      }
      width={24}
      xml="<g />"
    />
  </View>
</View>
`;

exports[`renders the icon in a touchable opacity 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    accessibilityLabel="Label"
    accessible={true}
    focusable={false}
    isTVSelectable={true}
    style={
      Object {
        "alignContent": "center",
        "alignItems": "center",
        "opacity": 1,
      }
    }
  >
    <SvgXml
      height={24}
      style={
        Object {
          "color": "#4051DB",
        }
      }
      width={24}
      xml="<g />"
    />
  </View>
</View>
`;
